### PR TITLE
Story lvjavs 970 – abonnements, added base pagination

### PR DESCRIPTION
**Pull request by Yevhensh**
1. Added pagination to front-end, jquery datatable.
2. Added back end classes that refers to jquery datatable queries: ../dto/pagination/*
3. Added some methods to persistence layer.
4. Added future class AbonnementsStatistics, now with one method, which will help in future in math or another kind of operations.
5. Add some validations to front-end with jquery validate.
6. Add some validations to back-end, but not use it yet.
7. Implementing some of crud methods.
8. Refactoring.

### CODE REVIEW

File: src/main/java/ua/softserveinc/tc/controller/manager/ManagerReportController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package ua.softserveinc.tc.controller.manager;
```
Comment (Yevhensh) [inline src/main/java/ua/softserveinc/tc/controller/manager/ManagerReportController.java]: sorry this class was removed with previous commits

File: src/main/webapp/resources/js/admin-abonnements.js [hunk 2]
```diff
@@ -1,7 +1,206 @@
 'use strict';
```
Comment (Yevhensh) [inline src/main/webapp/resources/js/admin-abonnements.js]: this file that copies previous will be removed in future

File: pom.xml [hunk 3]
```diff
@@ -474,6 +474,12 @@
             <scope>test</scope>
         </dependency>
 
+        <!-- MODEL MAPPER -->
+        <dependency>
+            <groupId>org.modelmapper</groupId>
+            <artifactId>modelmapper</artifactId>
+            <version>0.7.7</version>
+        </dependency>
```
Comment (Tat0) [inline pom.xml]: There are some problems with this model mapper. And I`ve removed it from the project

File: src/main/webapp/resources/js/admin-abonnement.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                        $(nTd).html(
+                            "<span class='name'>" + sData + "</span>"
+                        );
+                    }
+                },
+                {
+                    'data': 'price',
+                    'fnCreatedCell': function (nTd, sData) {
+                        $(nTd).html(
+                            "<span class='price'>" + sData + "</span>"
```
Comment (Tat0) [inline src/main/webapp/resources/js/admin-abonnement.js]: the same append

File: pom.xml (hunk 3 above)
Comment (Yevhensh) [inline pom.xml]: i'll remove it later

**Pull request merged**